### PR TITLE
Data sync merge

### PR DESCRIPTION
**Pull request by adambabik**
### Changes:

- `DataSyncWhisperAdapter` was created in `adapter/`. It conforms to `protocol.Protocol` just like other adapters. It handles incoming packets in Whisper messages by passing them to an object implementing `PacketHandler` interface.
- There is a new package `protocol/datasync` which contains `DataSyncNodeTransport` which implements an interface required by `mvds.Node`.
- Additionally, there is `protocol/datasync/peer` which contains data sync peer utility functions. It's in a separate package to avoid cyclic import (should probably be refactored and some code moved to `mvds`).

To run data sync use `-ds` flag.

### CODE REVIEW

**Comment by decanus** [comment]
@adambabik should we not merge this into `data-sync` rather than `master`?

File: protocol/adapter/datasync_whisper.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+				m.SigPubKey = item.Src
+				messages <- m
+			}
+		}
+	}()
+
+	return sub, nil
+}
+
+func (w *DataSyncWhisperAdapter) decodePayload(message *whisper.ReceivedMessage) (payload protobuf.Payload, err error) {
```
Comment (decanus) [inline protocol/adapter/datasync_whisper.go]: I guess this is fine, I am not usually a fan of named return variables though

Comment (adambabik) [reply protocol/adapter/datasync_whisper.go]: Me neither but in this case it works great :)

**Comment by adambabik** [comment]
> should we not merge this into data-sync rather than master?

@decanus no, I would prefer to keep just one branch. Initially, I suggested creating `data-sync` branch just because we need to refactor the code to support something like data sync. But as it happened in #84, I was able to move code from `data-sync` to this PR and merge it to master without affecting anything else or duplicating code.

For other PRs, I suggest we merge this one and then rebase these other branches on master. I know that there will be many conflicts but I don't see other way... Having just one branch is much more beneficial than one time conflict resolution.

**Comment by decanus** [comment]
@adambabik ok sounds good

**Pull request merged**